### PR TITLE
Fix Creating and Deleting data folders

### PR DESCRIPTION
**Pull request by christsim**
When clearing a data folder, it will only delete the contents if it exists, but it won't remove the root folder.  - This is to support mounting of those folders into a different location on the disk and for Docker volumes.

Similarly, when creating a folder, it won't fail if it already exists.

### CODE REVIEW

File: Communication/whisperNetwork.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 var async = require('async');
+var path = require('path');
 
 var events = require('../eventEmitter.js');
 var util = require('../util.js');
-var config = require('../config.js')
-var ports = require('../config.js').ports
+var config = require('../config.js');
+let ports = config.ports;
+let setup = config.setup;
```
Comment (coeniebeyers) [inline Communication/whisperNetwork.js]: Any reason for them to be `let`s instead of `var`s?

File: config.js [hunk 2]
```diff
@@ -30,6 +31,8 @@ config.whisper.powTarget = 0.5
 
 // Change these for different setups. 
 config.setup = {}
+// data dir
+config.setup.dataDir = env.DATA_DIR ? path.join(env.DATA_DIR, '/').replace(/\/$/, '') : '.';
```
Comment (coeniebeyers) [inline config.js]: This has been loooong overdue in the QNM, thanks!

File: util.js [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
+var prompt = require('prompt');
+var path = require('path');
 prompt.start();
 
-let config = require('./config.js')
-let ports = config.ports
-let setup = config.setup
+let config = require('./config.js');
+let ports = config.ports;
+let setup = config.setup;
```
Comment (coeniebeyers) [inline util.js]: You like your end lines :) I've slowly been removing them where I work on the code base, hence the inconsistency

**Comment by coeniebeyers** [comment]
Also, I noticed that `npm test` consistently fails with 
```
genesisConfigHandler readFile ERROR: { Error: ENOENT: no such file or directory, open 'quorum-genesis.json'
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: 'quorum-genesis.json' }
```
Please take a look at this as well. My assumption is that something isn't getting set correctly in `test/dynamicRaftMembership.js` where node 1 needs to serve the quorum-genesis.json to node 2.

File: Communication/whisperNetwork.js (hunk 1 above)
Comment (christsim) [reply Communication/whisperNetwork.js]: Let's were a copy and paste from one of your other files.

File: config.js (hunk 2 above)
Comment (christsim) [reply config.js]: didn't realize the datadir stuff was in this pull request, was still busy with it.  Still want to move all the folders like Blockchain etc. into the datadir, but haven't gotten round to it.

File: util.js (hunk 3 above)
Comment (christsim) [reply util.js]: Ya, sorry.  I'm using google coding standard, so it's just a habit for me now: https://google.github.io/styleguide/jsguide.html#disallowed-features-automatic-semicolon-insertion.  I can remove if u wish.

**Comment by christsim** [comment]
The npm tests are failing because of the changing of current directory.  Let me have a look how we can solve this.

**Comment by christsim** [comment]
I'll create a new pull request once the npm tests are working.

**Pull request closed without merging**